src/Security/SecurityIdentifier.cs

```diff
@@ -13,6 +13,26 @@ public sealed class SecurityIdentifier
 
     public string Value => ToString();
 
+    /// <value>
+    /// The domain SID of the identifier
+    /// </value>
+    public SecurityIdentifier AccountDomainSid
```

Comment:
We can return `null` here so the return type should be nullable.

```suggestion
    public SecurityIdentifier? AccountDomainSid
```